crates/network-primitives/src/traits.rs

```diff
@@ -76,12 +88,42 @@ pub trait TransactionResponse {
     /// Transferred value
     fn value(&self) -> U256;
 
+    /// Gas Price
+    fn gas_price(&self) -> Option<u128>;
+
     /// Gas limit
     fn gas(&self) -> u128;
 
+    /// Max BaseFeePerGas the user is willing to pay
+    fn max_fee_per_gas(&self) -> Option<u128>;
+
+    /// The miner's tip
+    fn max_priority_fee_per_gas(&self) -> Option<u128>;
+
+    /// Configured max fee per blob gas for eip-4844 transactions
+    fn max_fee_per_blob_gas(&self) -> Option<u128>;
+
     /// Input data
     #[doc(alias = "calldata")]
     fn input(&self) -> &Bytes;
+
+    /// Transaction signature
+    fn signature(&self) -> Option<Signature>;
```

Comment:
@emhane `Signature` comes from `alloy-rpc-types-eth` which can't be imported `network-primitives/Cargo.toml` due to a cyclic dependency since `rpc-types-eth/Cargo.toml` already pulls in network-primitives.

Do you have a suggestion here?

Reply:
probably best to add signature as associated type to the trait